features/step_definitions/operators.rb

```diff
@@ -160,3 +160,35 @@
   cb_name = 'operator_channel_name' unless cb_name
   cb[cb_name] = cluster_version('version').channel.split('-')[1]
 end
+
+Given /^I wait(?: up to (\d+) seconds)? for operator "([^"]*)" to check conditions: "([^"]*)"$/ do | time, operator, conditions |
+  ensure_admin_tagged
+
+  interval_time = 15
+  expected = {}
+  # Parse the conditions to Hash value {Available"=>"True", "Progressing"=>"False", "Degraded"=>"False", "Upgradeable"=>"True"}
+  if conditions.include? "="
+    conditions.split(/\s+/).map{|ar| ar.split('=')}.each {|k, v| expected[k]=v}
+  else
+    arr_conditions = conditions.split(/\s+/)
+    ["Available", "Progressing", "Degraded", "Upgradeable"][0,arr_conditions.length].zip(arr_conditions).map {|k, v| expected[k]=v}
+  end
+  raise "### Without valid conditions, unable to continue to check !!!" unless expected.length >0
+
+  time = Integer(time) rescue 60
+  interval_time = 30 if time > 100
+  actual_results = {}
+  success = wait_for(time, interval: interval_time) {
+    runtime_conditions = cluster_operator(operator).conditions
+    runtime_conditions.each do |c|
+      # only care about the `expected`, ignore others
+      if expected.keys.include? c['type']
+        actual_results[c['type']] = cluster_operator(operator).condition(type: c['type'], cached: false)['status']
+      end
+    end
```

Comment:
You should iterate `expected` instead of `runtime_conditions`:
`expected.each do | c |`

Reply:
Because` wait_for` with interval time, `runtime_conditions` is current conditions of operator every attempt , `expected` is the  comparison value that we got from input parameter `conditions`.

Reply:
No. If `oc get clusteroperator the_operator -o yaml` has 20 condition names, runtime_conditions will have 20 conditions, runtime_conditions.each will iterate 20 times.

Reply:
`runtime_conditions` got  all the conditions of operator every attempt, then iterate the every condition only matches included in `expected` ({Available"=>"True", "Progressing"=>"False", "Degraded"=>"False", "Upgradeable"=>"True"}), see following line,
`if expected.keys.include? c['type']`